java-desktop/src/main/java/com/couchbase/todo/model/DB.java

```diff
@@ -246,25 +247,33 @@ private void startReplication(@NotNull String username, @NotNull String password
         final URI sgUri = getReplicationUri();
         if (sgUri == null) { return; }
 
+        final int attempts = TodoApp.getConfig().getAttempts();
```

Comment:
Why have variables for these things?  In general, I think that variables make things more complex: their values might change.  Why not just use the full function call when you set up the replicator?